docs/website/docs/dlt-ecosystem/verified-sources/scrapy.md

```diff
@@ -0,0 +1,188 @@
+   URL: "https://quotes.toscrape.com/page/1/". 
+
+   ```python
+   class MySpider(Spider):
+    def parse(self, response: Response, **kwargs: Any) -> Any:
+        for next_page in response.css("li.next a::attr(href)"):
+            if next_page:
+                yield response.follow(next_page, self.parse)
+
+        for quote in response.css("div.quote"):
+            result = {
+                "quote": {
+                    "text": quote.css("span.text::text").get(),
+                    "author": quote.css("small.author::text").get(),
+                    "tags": quote.css("div.tags a.tag::text").getall(),
+                },
+            }
+            yield result
+   ```
+
+   > Define your own class tailored to the website you intend to scrape. 
+
+1. Configure the pipeline by specifying the pipeline name, destination, and dataset as follows:
+
+   ```python
+   pipeline = dlt.pipeline(
+       pipeline_name="scrapy_pipeline",  # Use a custom name if desired
+       destination="duckdb",  # Choose the appropriate destination (e.g., duckdb, redshift, post)
+       dataset_name="scrapy_data",  # Use a custom name if desired
+   )
+   ```
+
+   To read more about pipeline configuration, please refer to our
+   [documentation](../../general-usage/pipeline).
+
+1. To run the pipeline with customized scrapy settings:
+
+   ```python
+   run_pipeline(
+        pipeline,
+        MySpider,
+        # you can pass scrapy settings overrides here
+        scrapy_settings={
+            # How many sub pages to scrape
+            # https://docs.scrapy.org/en/latest/topics/settings.html#depth-limit
+            "DEPTH_LIMIT": 100,
+            "SPIDER_MIDDLEWARES": {
+                "scrapy.spidermiddlewares.depth.DepthMiddleware": 200,
+                "scrapy.spidermiddlewares.httperror.HttpErrorMiddleware": 300,
+            },
+            "HTTPERROR_ALLOW_ALL": False,
+        },
+        write_disposition="append",
+    )
+   ```
+
+   In the above example, scrapy settings are passed as a parameter. For more information about
+   scrapy settings, please refer to the
+   [Scrapy documentation.](https://docs.scrapy.org/en/latest/topics/settings.html).
+
+1. To limit the number of items processed, use the "on_before_start" function to set a limit on 
+   the resources the pipeline processes. For instance, setting the resource limit to two allows 
+   the pipeline to yield a maximum of two resources.
+
+   ```python
+   def on_before_start(res: DltResource) -> None:
+        res.add_limit(2)
+
+    run_pipeline(
+        pipeline,
+        MySpider,
+        batch_size=10,
+        scrapy_settings={
+            "DEPTH_LIMIT": 100,
+            "SPIDER_MIDDLEWARES": {
+                "scrapy.spidermiddlewares.depth.DepthMiddleware": 200,
+                "scrapy.spidermiddlewares.httperror.HttpErrorMiddleware": 300,
+        },},
+        on_before_start=on_before_start,
+        write_disposition="append",
+    )
+   ```
```

Comment:
what's going on with spaces in the code snippets? :D
David works on auto linting now, but we still should be careful 
![image](https://github.com/dlt-hub/dlt/assets/20367975/d840dc64-de49-4cb4-b102-43017ed0095b)

Reply:
hehe, yes, this will be caught :) it's on devel already 🥸

Reply:
updated, thanks!